Updated the django-countries to support Django 2.2

mraarif opened the pull request:
Relevant JIRA ticket can be found [here](https://openedx.atlassian.net/browse/BOM-1082).

Few countries have title changes. [Complete details.](https://github.com/SmileyChris/django-countries/commits/master/django_countries/data.py)

awais786 commented on common/djangoapps/student/tests/test_user_profile_properties.py:
```diff
@@ -90,7 +90,7 @@ def test_display_gender_none_set(self):
     def test_invalidate_cache_user_profile_country_updated(self):
 
-        country = 'us'
+        country = 'US'
```
Why you changed this ?

awais786 commented on openedx/core/djangoapps/user_authn/views/tests/test_register.py:
```diff
@@ -302,7 +303,7 @@ class RegistrationViewTest(ThirdPartyAuthTestMixin, UserAPITestCase):
     ADDRESS = "123 Fake Street"
     CITY = "Springfield"
-    COUNTRY = "us"
+    COUNTRY = "US"
```
same question ?

awais786 commented on openedx/core/djangoapps/user_authn/views/tests/test_register.py:
```diff
@@ -548,7 +549,7 @@ def test_register_form_third_party_auth_running_google(
                 {
                     "value": country_code,
-                    "name": six.text_type(country_name),
+                    "name": country_name,
```
why you removed `six.text_type`?

awais786 commented on openedx/core/djangoapps/user_authn/views/tests/test_register.py:
```diff
@@ -1600,15 +1601,27 @@ def _assert_fields_match(self, actual_field, expected_field):
+        actual_options = actual_field.get("options")
+        expected_options = expected_field.get("options")
+
+        if actual_options and expected_options:
```
why adding if else in test ?

Ayub-Khan replied:
I think we might want to change our implementations instead of changing tests. To make sure it's working the same.

Ayub-Khan commented on openedx/core/djangoapps/user_authn/views/tests/test_register.py:
```diff
@@ -1600,15 +1601,27 @@ def _assert_fields_match(self, actual_field, expected_field):
                 )
-            )
+
+    def _test_countries(self, expected, actual):
```
use the dict comprehensions to generate a dict with all members of dict one present in 2 and than compare one with 2 that using assertEqual.

mraarif commented:
jenkins run py35-django111 bokchoy

jmbowman commented:
Unfortunately, it looks like they didn't start testing with Django 2.2 until version 5.4.  Also, the capitalization change sounds like a bug fix; the official ISO 3166 country codes are all upper case.  How extensive are the code changes needed to use the latest release?

awais786 commented:
> Unfortunately, it looks like they didn't start testing with Django 2.2 until version 5.4. Also, the capitalization change sounds like a bug fix; the official ISO 3166 country codes are all upper case. How extensive are the code changes needed to use the latest release?

@jmbowman code changes are in tests cases but main concern was caps issue, if we are storing values in db as small ? earlier in few tests @mraarif changed the values from small to cap.

mraarif commented:
@jmbowman we've switched back to the latest version of `django-countires` aside from capitalization change ISO names of the following countries have been changed

1. Macedonia to  North Macedonia
2. Swaziland to Eswatini

and we're assuming the system is only saving two-letter country codes in DB so with the new version of `django-countries` updated names of these countries will be displayed (as the country codes remained the same) I've updated the constants these updated names.

awais786 commented:
> @jmbowman we've switched back to the latest version of `django-countires` aside from capitalization change ISO names of the following countries have been changed
> 
> 1. Macedonia to  North Macedonia
> 2. Swaziland to Eswatini
> 
> and we're assuming the system is only saving two-letter country codes in DB so with the new version of `django-countries` updated names of these countries will be displayed (as the country codes remained the same) I've updated the constants these updated names.

https://github.com/SmileyChris/django-countries/commits/master/django_countries/data.py
